crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -576,23 +631,22 @@ pub fn mesh_resource_provider_system(
             }
         }
 
-        if let Some(RenderResourceId::Buffer(vertex_buffer)) =
-            render_resource_context.get_asset_resource(*handle, VERTEX_BUFFER_ASSET_INDEX)
-        {
-            render_pipelines.bindings.set_vertex_buffer(
-                "Vertex",
-                vertex_buffer,
-                render_resource_context
-                    .get_asset_resource(*handle, INDEX_BUFFER_ASSET_INDEX)
-                    .and_then(|r| {
-                        if let RenderResourceId::Buffer(buffer) = r {
-                            Some(buffer)
-                        } else {
-                            None
-                        }
-                    }),
-            );
+        // set index buffer into binding
+        let index_buffer_resource = render_resource_context.get_asset_resource(*handle, INDEX_BUFFER_ASSET_INDEX)
+            .expect("no index buffer resource found.").get_buffer().expect("resource is not a buffer");
+
+        render_pipelines.bindings.set_index_buffer(index_buffer_resource);
+
+        // set vertex buffers into bindings
+        for index in 1..4 { //TODO: use actual vertex buffer count
+            if let Some(RenderResourceId::Buffer(vertex_buffer)) =
+                render_resource_context.get_asset_resource(*handle, index)
+            {
+                println!("setup vertex buffer {}", index);
+                render_pipelines.bindings.set_vertex_buffer(index as u8, vertex_buffer);
+            }
```

Comment:
based on my above comments, this would end up as something like

```rust
for pipeline in render_pipelines.pipelines {
    let pipeline_descriptor = pipeline_descriptors.get(pipeline.descriptor).unwrap();
    if let Some(layout) = pipelin_descriptor.layout {
        for (slot, vertex_buffer_descriptor) in layout.vertex_buffer_descriptors.iter().enumerate() {
            let handle = render_resource_context
                .get_asset_resource(*handle, vertex_buffer_descriptor.name)
                .expect("mesh does not provide required vertex attribute {} for pipeline {:?}", vertex_buffer_descriptor.name, pipeline_descriptor);
            println!("setup vertex buffer {}", slot);
            render_pipelines.bindings.set_vertex_buffer(slot as u8, handle);
        }
    }
}
```